lib/gpu/lib/src/shader_library.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: public_member_api_docs
+
+part of flutter_gpu;
+
+base class ShaderLibrary extends NativeFieldWrapperClass1 {
+  static ShaderLibrary? fromAsset(String assetName) {
+    final lib = ShaderLibrary._();
+    final error = lib._initializeWithAsset(assetName);
+    if (error != null) {
+      throw Exception("Failed to initialize ShaderLibrary: ${error}");
+    }
+    return lib;
+  }
+
+  ShaderLibrary._();
+
+  Shader? operator [](String shaderName) {
```

Comment:
nit: use a method for this. operators, getters, and `[]` feel cheap even when they aren't.

Reply:
I think this is sufficiently cheap? It just grabs a handle to an already existing object on the C++ side.